src/compiler.ts

```diff
@@ -8693,7 +8693,14 @@ export class Compiler extends DiagnosticEmitter {
             (<FunctionDeclaration>baseCtor.declaration).clone()
           ),
           null,
-          baseCtor.signature,
+          Signature.create(
+            this.program,
+            baseCtor.signature.parameterTypes,
+            classInstance.type,
+            classInstance.type,
+            baseCtor.signature.requiredParameters,
+            baseCtor.signature.hasRest
+          ),
```

Comment:
Perhaps better reuse `clone` instead `Signature.create`?
```ts
baseCtor.signature.clone()
```

Reply:
`thisType` and `returnType` is readonly field. These can only be modified during constructor. `clone` is simply copy a same `Signature`

Reply:
I see